io.github.ctlcltd.e2se.yaml

```diff
@@ -0,0 +1,43 @@
+app-id: io.github.ctlcltd.e2se
+runtime: org.kde.Platform
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: e2-sat-editor
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=x11
+  - --socket=cups
+  - --filesystem=host
```

Comment:
This software manages channel lists, it can be a tool/utility. It assumes a free use of the filesystem, it shouldn't be too strict.

Reply:
For you "not too strict" is "free for all"?

Reply:
since this uses Qt, it likely uses the file portals. so you might not have any need for the filesystem access.

Reply:
This is written in the documentation "--filesystem=host - **access normal files on the host**, not including host os or system internals described below". it's a channellist management tool, you need access to the filesystem.

Reply:
I have no idea what a "channellist management tool" is.

Reply:
There is a problem with portals, the Flathub installation guides don't show portals, it means that a normal user should have more knowledge about these things. If it's not a serious problem, I'd prefer a normal use of the filesystem, as accepted by the current documentation. In short, if a user installs my flatpak and doesn't see the files it becomes unusable.

Reply:
> I have no idea what a "channellist management tool" is.

It's a software that manages satellite, cable, terrestrial channel lists, dreambox-like decoder lists. This software connects to the decoder via FTP and telnet. The question is: Does Flathub accept software that uses the following configuration: "--filesystem=host"? If the answer is no, I'll try to follow the guidelines for an Alcatraz-style use 🙂.

Reply:
@hfiguiere It works without "--filesystem=host". But it works on my Arch Linux with Xfce4 and **xdg-desktop-portal** installed. Works with problems... by default install, I see that in language other than english it shows me all squares instead of text in file chooser dialog, I had to change it from language to english and back. I also see the whole filesystem, I don't know if this sandbox works like this.

I understand the need, this issue: https://github.com/flathub/flathub/issues/1337 and this: https://github.com/search?q=org%3Aflathub+%22filesystem%3Dhost%22&type=Code&p=1

I don't know if it will work with xdg-desktop-portal on all distros supported by Flathub. I don't understand such strictness: *"The Flatpak sandbox is still a work in progress."* as written here: https://github.com/flathub/flathub/wiki/App-Requirements

Reply:
@hfiguiere I did some tests. My software saves more than one file at a time, channel lists are organized in folders (folder with multiple files inside) and in some circumstances: Export all to HTML or CSV. Features that are currently not supported by **xdg-desktop-portal**. As documented here: https://docs.flatpak.org/en/latest/portals.html#portal-support-in-qt-and-kde *"Note that portals cannot currently give access to directories on the host filesystem"*. And related issues: https://github.com/flatpak/xdg-desktop-portal/issues/997 and https://github.com/flatpak/xdg-desktop-portal/issues/973. I would gladly remove "--filesystem=host" from the permissions but at the moment it is not possible, due to the **xdg-desktop-portal** status.